src/magic.cpp

```diff
@@ -1070,11 +1070,18 @@ std::string spell::damage_type_string() const
     return name_by_dt( dmg_type() );
 }
 
+#ifndef __aarch64__
 // constants defined below are just for the formula to be used,
 // in order for the inverse formula to be equivalent
 constexpr float a = 6200.0f;
 constexpr float b = 0.146661f;
 constexpr float c = -62.5f;
+#else
+// ARM64 needs double
+constexpr double a = 6200.0;
+constexpr double b = 0.146661;
+constexpr double c = -62.5;
+#endif
```

Comment:
```suggestion
// constants defined below are just for the formula to be used,
// in order for the inverse formula to be equivalent
constexpr double a = 6200.0;
constexpr double b = 0.146661;
constexpr double c = -62.5;
```
agreed with ZS, you yourself said some architectures fail out at level 47